membership-attribute-service/app/components/TouchpointComponents.scala

```diff
@@ -143,20 +143,20 @@ class TouchpointComponents(
   )
 
   lazy val futureCatalog: Future[CatalogMap] = catalogService.catalog
-    .map(_.fold[CatalogMap](error => { println(s"error: ${error.list.toList.mkString}"); Map() }, _.map))
+    .map(_.fold[CatalogMap](error => { logger.errorNoPrefix(scrub"error: ${error.list.toList.mkString}"); Map() }, _.map))
     .recover { case error =>
-      logger.error(scrub"Failed to load the product catalog from Zuora due to: $error")
+      logger.errorNoPrefix(scrub"Failed to load the product catalog from Zuora due to: $error")
       throw error
     }
 
   lazy val subscriptionService: SubscriptionService = {
-    lazy val zuoraSubscriptionService = new ZuoraSubscriptionService(productIds, futureCatalog, zuoraRestClient, zuoraSoapService.getAccountIds)
+    lazy val zuoraSubscriptionService = new ZuoraSubscriptionService(productIds, futureCatalog, zuoraRestClient, zuoraSoapService)
 
     subscriptionServiceOverride.getOrElse(
       new SubscriptionServiceWithMetrics(zuoraSubscriptionService, createFineMetrics),
     )
   }
-  lazy val paymentService: ZuoraPaymentService = new ZuoraPaymentService(zuoraSoapService, catalogService.unsafeCatalog.productMap)
+  lazy val paymentService: PaymentService = new PaymentService(zuoraSoapService, catalogService.unsafeCatalog.productMap)
```

Comment:
this had a trait and a class but there was only one implementation, so I squashed them together for now.